### PR TITLE
Add spice module to documentation

### PR DESCRIPTION
**Pull request by tmplummer**
# Change Summary

## Overview
These are all changes that were needed to add the spice directory to the documentation build and fix some formatting issues.

### CODE REVIEW

File: docs/source/conf.py
Comment (tmplummer) [inline docs/source/conf.py:120]: More nitpick ignores for `IntEnum`s and some `numpy_typing`. What confuses me is that theses sphinx warnings would go away if I built the documentation a second time without cleaning. I simply couldn't figure out how to fix these issues without just adding the ignore regex here.

File: imap_processing/spice/geometry.py
Comment (tmplummer) [inline imap_processing/spice/geometry.py:99]: I have no clue why sphinx was having problems with this `optional` here, which adheres to `numpydoc` syntax. I spent way too long trying to fix it but ended up just having to remove it.

**Review by laspsandoval — COMMENTED** [review]
I actually don't see anything that needs to be changed. Looks good to me, but I'll let other reviewers chime in before I hit "approve".

**Review by maxinelasp — APPROVED** [review]
LGTM. Sphinx is... its own challenge... so nice job integrating this

**Pull request merged**
